pkg/maps/multicast/subscribermap.go

```diff
@@ -182,6 +182,36 @@ func (m GroupV4OuterMap) Delete(group netip.Addr) error {
 }
 
 func (m GroupV4OuterMap) List() ([]netip.Addr, error) {
+	res, err := m.ListBatch()
```

Comment:
Wouldn't it be a bit better if we check this immediately after the map is OpenOrCreated and cache the result? 

Then, we don't need to do an extraneous ListBatch check on every list.

Reply:
Yeah, I was debating about this in my mind. From code change prospective this looked least intrusive. Let me change the approach and cache the result upfront.

Reply:
Hi @ldelossa, going over this again. Here is proposed change where we lookup if batch operation is allowed at map creation/lookup time. 
https://gist.github.com/harsimran-pabla/1c0f04675f884227aff40f8b945e527a

There are couple of caveats, not sure if change required or not.

Whenever we get map from kernel ( lookup in group map to get subscriber map ), or in CLI to get group map. We will need to do batch lookup to see if it is supported or not and cache result in map structure which is returned. 

In case of group list operation caching it makes sense. In subscriber map, we need to do this operation on each subscriber lookup - so we add extra syscall to group.Lookup().  Overall number of syscalls is unchanged

Here is number of extra syscalls we introduce in caching method

1. Get group map ( 1 syscall )
2. Get list of subscribers
3. Lookup subscriber ( 1 syscall per subscriber )
4. List subscriber 

vs extra syscall per List operation ( in original PR )

1. Get group map
2. Get list of subscribers ( 1 syscall )
3. Lookup subscriber
4. List subscriber ( 1 syscall per subscriber )

Only place where we save extra syscall is if caller stores group map ( step 1 in caching method ).

Do you still think we should go with caching strategy ?

Reply:
So did some more thinking. Actually, we can rely on BatchLookup in subscriber map ( type map.hash ). Minimum kernel version required for multicast is 5.13, in which batch lookup for map type Hash is supported. We don't really need to check for its support. 

Only check is required is for HashOfMap type, which is group map. We do it once at startup and each time when group CLI is called.